src/metabase/public_settings.clj

```diff
@@ -599,3 +599,24 @@
                     (not (t2/exists? 'Database :is_sample false, :initial_sync_status "complete"))
                     ;; frontend should set this value to `true` after the modal has been shown once
                     v))))
+
+(defsetting uploads-enabled
+  (deferred-tru "Whether or not csv uploads are enabled")
+  :visibility :admin
+  :type       :boolean
+  :default    false)
```

Comment:
Note if this is disabled, does `uploads-database-id`, `uploads-schema-name`, and `uploads-table-prefix` can be nil. I don't think this is a problem as long as we don't assume that if they are non-nil then `uploads-enabled` is true.